Dockerfile.konflux

```diff
@@ -0,0 +1,57 @@
+
+#ARG BUILD_BASE=develop
+
+#go-1.21
+FROM registry.redhat.io/ubi8/go-toolset:1.21@sha256:742ae6ec1aef3e7faae488c47695fb64964d342aefecf52d23bd9d5e6731d0b6 AS build
+
+USER root
+
+# needed for konflux as the previous stage is not used
+WORKDIR /opt/app
+COPY go.mod go.sum ./
+# Download dependencies before copying the source so they will be cached
+RUN go mod download
+
+# Copy the source
+COPY . ./
+
+USER root
+
+# Build the binaries using native go compiler from BUILDPLATFORM but compiled output for TARGETPLATFORM
+# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+RUN go build -o puller model-serving-puller/main.go && \
```

Comment:
can we do the same as here? https://github.com/red-hat-data-services/modelmesh-runtime-adapter/blob/rhoai-2.13/Dockerfile.konflux#L23